### DISCUSSION TITLE
supported versions fast login?

### DISCUSSION THREAD

**Discussion by Utlamo** [games647/FastLogin#569] [Q&A] [answered]
For which MC version does the plugin work?
I try use plugin for spigot 1.16.5 and always premium player has cracked UUID.
Where is latest build? How i can get latest build(assuming I downloaded wrong)

I try use plugin for Protolib, LuckPerms, and login security - dont work.
Try for another login plugin. Don't work too.
Sory for my bad english and I hope you will understand what I am writing

**Comment by TuxCoding** [comment, marked as answer]
>For which MC version does the plugin work?

1.8.8-1.17

>I try use plugin for spigot 1.16.5 and always premium player has cracked UUID.

The plugin will only set the premium UUID if the login process is authenticated through Mojang. By default the plugin always assumes connecting players are cracked. They can opt-in using the command option `/premium`. For new players it's also possible to enable `autoRegister`, so that the plugin assumes a paid player based on the username.

The background for this behavior is that otherwise cracked players will be disconnected as it appears if the server would be in onlinemode. This could potentially lock out users or create confusing situations. You can read a little bit more about it here: https://github.com/games647/FastLogin/wiki/FAQ#why-do-players-have-to-invoke-a-command

>Where is latest build?

Projects builds are released here: https://ci.codemc.io/job/Games647/job/FastLogin/
Depending on the platform, you need to download the version under last successful artifact and put it into the plugins folder.
